src: fix process exit listeners not receiving unsettled tla codes (#56872)
nodejs/node: doc/api/process.md

exits gracefully, or is exited via [`process.exit()`][] without specifying
a code.

> The value of `process.exitCode` can be updated by either setting the value
> directly (e.g. `process.exitCode = 9`), or by specifying a code to
> [`process.exit(code)`][`process.exit()`] (the latter takes
> precedence and will override any previous setting of `process.exitCode`).

Reviewer: I wonder if it wouldn't be better to give examples instead

````suggestion
The value of `process.exitCode` can be updated by either assigning a value to
`process.exitCode` or by passing an argument to [`process.exit()`][]:

```console
$ node -e 'process.exitCode = 9'; echo $?
9
$ node -e 'process.exit(42)'; echo $?
42
$ node -e 'process.exitCode = 9; process.exit(42)'; echo $?
42
```
````

Author: Sounds good to me 🙂 :+1:

exits gracefully, or is exited via [`process.exit()`][] without specifying
a code.

The value of `process.exitCode` can be updated by either assigning a value to
`process.exitCode` or by passing an argument to [`process.exit()`][]:

```console
$ node -e 'process.exitCode = 9'; echo $?
9
$ node -e 'process.exit(42)'; echo $?
42
$ node -e 'process.exitCode = 9; process.exit(42)'; echo $?
42
```

The value can also be set implicitly by Node.js when unrecoverable errors occur (e.g.
such as the encountering of an unsettled top-level await). However explicit

> manipulations of the exit code always take precedence over implicit ones (meaning that
> for example updating `process.exitCode` to `9` guarantees that the program's exit code
> will be `9` regardless on other errors Node.js might encounter).

Reviewer: Same here
````suggestion
manipulations of the exit code always take precedence over implicit ones:

```console
$ node --input-type=module -e 'await new Promise(() => {})'; echo $?
13
$ node --input-type=module -e 'process.exitCode = 9; await new Promise(() => {})'; echo $?
9
```
````
